### PR TITLE
User Routes And User model

### PR DESCRIPTION
**Pull request by gilbertosantana24**
.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -60,6 +60,7 @@ app.get('/logout', (req, res) => {
 })
 
 app.use("/api", require("./routes"));
+app.use("/users/user", require("./userRoutes"));
```
Comment (jesushzv) [inline server.js:63]: No need to repeat /user/users, that creates unnecessary paths. /users is sufficient

File: userRoutes.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+const router = require('express').Router();
+const User = require('./userModel');
+
+//GET USERS
+router.get('/users', async (req,res)=> {
+    const user = await User.find();
```
Comment (jesushzv) [inline userRoutes.js:6]: Could rename "user" to "users", since we are trying to get all of the users in this endppoint

File: server.js (hunk 1 above)
Comment (jesushzv) [inline server.js]: There is no need for /users/user, this will create extra layers of routes. /users is sufficient.

File: userRoutes.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    res.send(user)
+})
+
+//GET USERS BY ID
+router.get('/:id', async (req,res)=> {
+    const user = await User.findById();
```
Comment (jesushzv) [inline userRoutes.js]: User.findById() takes no parameter, there's no way for the DB to retrieve the user for that specific id.

File: userRoutes.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    res.send(user);
+})
+
+
+//CREATE USER
+router.post('/createUser', async (req,res)=> {
+    const movie = await User.create({
```
Comment (jesushzv) [inline userRoutes.js]: Rename movie to user.  Probably a typo

File: userRoutes.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        name: req.body.name,
+        lastname: req.body.lastname,
+        username: req.body.username,
+        email: req.body.email,
+        password: req.body.password
+    });
+
+    res.send(movie);
```
Comment (jesushzv) [inline userRoutes.js]: We need to make sure that there are no repeated users or emails. Before we create the user, query the DB and check if there's already a username or email like the one provided. If there is, don't allow it.

**Review by jesushzv — CHANGES_REQUESTED** [review]
Requested some changes, leave a comment if further clarification is needed

File: server.js (hunk 1 above)
Comment (gilbertosantana24) [reply server.js]: Yes, there was an error from my end, good observation. Will fix

**Review by jesushzv — APPROVED** [review]
Reviewed and approved changes. Thanks a lot

**Pull request merged**